backend/src/resolvers/socialMedia.js

```diff
@@ -15,6 +18,11 @@ export default {
       )
       session.close()
 
+      return socialMedia
+    },
+    DeleteSocialMedia: async (object, params, context, resolveInfo) => {
+      const socialMedia = await neo4jgraphql(object, params, context, resolveInfo, false)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-03T10:35:56Z" title="Friday, May 3rd 2019, 12:35:56 pm +02:00">May 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -15,6 +18,11 @@ export default {
       )
       session.close()
 
+      return socialMedia
+    },
+    DeleteSocialMedia: async (object, params, context, resolveInfo) => {
+      const socialMedia = await neo4jgraphql(object, params, context, resolveInfo, false)
```


doesn't this create another `SocialMedia` node @Tirokk, or is there some magic that sees that you have called a delete action?

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-05-06T13:24:06Z" title="Monday, May 6th 2019, 3:24:06 pm +02:00">May 6, 2019</time>_

---


There seems to be a magic that deletes the node and with it the relation. @mattwr18 
Because it’s an autogenerated mutation.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-10T11:24:04Z" title="Friday, May 10th 2019, 1:24:04 pm +02:00">May 10, 2019</time>_

---


@mattwr18 you could inspect the generated cypher query of `neo4jgraphql` to get evidence, but I assume it's using `DELETE DETACH`. https://www.graphgrid.com/understanding-detach-delete-in-cypher/

Quite a final solution for a node and it's relationship. Comparable to Rails' `:dependent => :destroy`.